### PR TITLE
feat(profile) - bigquery: Fix for hitting limit with too many partitioned tables

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py [hunk 1]
```diff
@@ -748,7 +748,6 @@ def _generate_single_profile(
         custom_sql: str = None,
         **kwargs: Any,
     ) -> Optional[DatasetProfileClass]:
-        logger.info(f"Profiling {pretty_name}")
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:751]: intentional?

File: metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py [hunk 2]
```diff
@@ -94,12 +94,20 @@
     is_partitioning_column = 'YES'
     -- Filter out special partitions (https://cloud.google.com/bigquery/docs/partitioned-tables#date_timestamp_partitioned_tables)
     and p.partition_id not in ('__NULL__', '__UNPARTITIONED__')
+    and STORAGE_TIER='ACTIVE'
```
Comment (gabe-lyons) [inline metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:97]: [asking to learn] what does this do?

File: metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py (hunk 1 above)
Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py:751]: yep, I left here earlier non intential and now we duplicate this message

File: metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py (hunk 2 above)
Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:97]: It only list the active partitions and not those which are store for long term storage and even cost more to query.

Comment (treff7es) [reply metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py:97]: https://cloud.google.com/bigquery/docs/information-schema-tables#partitions_view

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**